Assisments/Assignment2/HexCalc.java

```diff
@@ -0,0 +1,372 @@
+	
+	/**
+	 * Gets the string.
+	 *
+	 * @return the string
+	 */
+	public static String getString(){
+		
+		while (true) {
+			try {
+					String str =sc.nextLine();
+					return str.replaceAll("[^a-zA-Z0-9]", "");
+				}
+			catch (Exception ex) {
+					System.out.println("Enter string value only :");
+					sc.next();
+				}
+			}
+	}
+	
+	/**
+	 * Gets the char.
+	 *
+	 * @param c the c
+	 * @return the char
+	 */
+	public static int getChar(char c) {
+		if( c >= '0' && c<= '9' ) {
+			return (c -'0');
+		}else{
+			return (c -'A'+10);
+		}
+	}
+	
+	/**
+	 * This method convert number to equivalent character in notation according to the base.
+	 *
+	 * @param num represent number which will be converted
+	 * @return the character in the respective notation
+	 */
+	public static char revChar(Integer num){
+		if(num >=0 && num <=9)
+			return (char) (num+'0');
+		else
+			return (char) (num + 'A' -10);
+	}
+	
+	/**
+	 * This method convert number of any base to decimal equivalent
+	 *
+	 * @param number represent the number with any base
+	 * @param base represent the base of number
+	 * @return the decimal equivalent of number
+	 */
+	public static Integer NBaseToDec(String number, Integer base) {
+		int len = number.length();
+		int power = 1;
+		Integer dec = 0 ;
+		
+		for(int iterator = len -1; iterator >= 0; iterator-- ) {
+			if( getChar(number.charAt(iterator)) >= base) {
+				return Integer.MIN_VALUE;
+			}
+			else{
+				dec += getChar(number.charAt(iterator)) * power;
+				power = power * base;
+			}
+		}
+		return dec;
+	}
+	
+	/**
+	 * This method convert decimal number to respective base
+	 *
+	 * @param dec represent the decimal number
+	 * @param base represent the base to convert to
+	 * @return the converted number in respective base
+	 */
+	public static String decToBaseN( Integer dec, Integer base){
+		String result ="";
+		while( dec > 0 ) {
+			result += revChar( dec % base );
+			dec /= base;
+ 		}
+		String decm = "";
+		char ch[] = result.toCharArray();
+	    for(int iterator = ch.length - 1 ; iterator >= 0 ; iterator-- ) {  
+	        decm += ch[iterator];  
+	    } 
+	    return decm;
+	    	
+	}
+	
+	/**
+	 * this method compare the length of the numbers
+	 *
+	 * @param num1 represent number to be compared
+	 * @param num2 represent number to be compared
+	 * @return the bigger number or equal if the numbers are equal
+	 */
+	public static String compareLength(String num1 , String num2){
+		int len1 = num1.length();
+		int len2 = num2.length();
+		if( len1 < len2 )
+			return num2;
+		else if( len1 > len2)
+			return num1;
+		else
+			return "equal";
+	}
+	
+	/**
+	 * This method compare numbers are equal or not
+	 *
+	 * @param num1 represent number to be compared
+	 * @param num2 represent number to be compared
+	 * @return true, if equal, else return false
+	 */
+	public static boolean compareEqual(String num1, String num2) {
+	
+		String bigNum = compareLength(num1, num2);
+		if( bigNum.equals(num1) )
```

Comment:
If conditional can be simplified here, can use OR operator for better performance rather then use if statements.